fix: correct typos in javascript-arrays-quiz and review-javascript files

jenishkatuwal11 opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or Gitpod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #61727

<!-- Feel free to add any additional description of changes below this line -->

This PR fixes typos and grammatical issues in:

- `curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md`
- `curriculum/challenges/english/25-front-end-development/review-javascript-arrays/6723c66f623701a3cdf72130.md`

**Changes:**
- Added missing word "to" in quiz question: changed "correct way access" to "correct way to access".
- Corrected “a array” → “an array”.
- Wrapped `-1` in backticks for `indexOf()`.
- Improved clarity for `slice()` method description.

This is a documentation-only change, no functional code was altered.

jenishkatuwal11 committed:
fix: correct typos in javascript-arrays-quiz and review-javascript files

jenishkatuwal11 commented:
Hi! This PR fixes several typos and minor grammatical errors in the `javascript-arrays-quiz` and `review-javascript` files. These changes aim to enhance the clarity and readability of the content for learners.

Please let me know if any further adjustments are needed. Thank you for reviewing my contribution!

Looking forward to your feedback.

ilenia-magoni commented:
As a new contributor, we encourage you to read our contributing guidelines. 

We expect our contributors to be aware of the process specific to this project. Following the guidelines religiously earns you the respect of fellow maintainers and saves everyone time.

Some examples of this are:

1. Make sure you are not editing off of GitHub (while you can, it's not a good idea).
2. Makes sure you actually follow the PR checklist and not just tick things off; we won't take you seriously otherwise.
3. Use the correct way to link PRs in the description of the PR by updating the `XXXXXX`.
4. ... and similar things.

We appreciate you taking the time to help us, and we hope to see more contributions from you.

Cheers & Happy Coding!

ilenia-magoni commented:
it looks like you deleted the template that is there when you create the PR, that template is an important part of making sure you followed the properprocedure, you should not delete it. I have added it in this post so you can restore it.

```md
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [ ] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [ ] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [ ] My pull request targets the `main` branch of freeCodeCamp.
- [ ] I have tested these changes either locally on my machine, or Gitpod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #XXXXX

<!-- Feel free to add any additional description of changes below this line -->
```

huyenltnguyen reviewed (commented):
Thank you for the PR.

Please avoid using @-mentions unless it is necessary. Tagging people directly can create extra notifications and may overwhelm maintainers. Thank you for your understanding.

jenishkatuwal11 committed:
Update curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccbba6dacdb65a59067.md

Co-authored-by: Huyen Nguyen <25715018+huyenltnguyen@users.noreply.github.com>

jenishkatuwal11 committed:
Update curriculum/challenges/english/25-front-end-development/review-javascript/6723d3cfdd0717d3f1bf27e3.md

Co-authored-by: Huyen Nguyen <25715018+huyenltnguyen@users.noreply.github.com>

jdwilkin4 commented on curriculum/challenges/english/25-front-end-development/review-javascript/6723d3cfdd0717d3f1bf27e3.md:
```diff
@@ -823,7 +823,9 @@ const newList = programmingLanguages.concat("Perl");
-- **`slice()` Method**: This method returns a shallow copy of a portion of the array at a specified index or the entire array. A shallow copy will copy the reference to the array instead of duplicating it.
+- **`slice()` Method**: This method returns a shallow copy of a portion of the array, starting from a specified index or the entire array. 
+A shallow copy creates a new array but does not deeply copy nested objects — if the array contains objects, their references are copied.
+
```
Also, revert this change in general as mentioned in this comment here.

https://github.com/freeCodeCamp/freeCodeCamp/issues/61727#issuecomment-3168839120

There will be a separate issue for that

jenishkatuwal11 committed:
fix: address reviewer feedback for quiz-javascript-arrays, review-javascript-arrays and review-javascript

- Corrected typos and grammar in:
  curriculum/challenges/english/25-front-end-development/quiz-javascript-arrays/66edcccba6dacdb65a59067.md
  curriculum/challenges/english/25-front-end-development/review-javascript-arrays/6723c66f623701a3cdf72130.md
  curriculum/challenges/english/25-front-end-development/review-javascript/6723d3cfdd0717d3f1bf27e3.md
- Updated slice() method description wording
- Wrapped -1 in backticks for indexOf() method
- Fixed “a array” to “an array”
- Removed unnecessary blank lines for clarity

jenishkatuwal11 committed:
Merge branch 'fix-typos-javascript-quiz' of https://github.com/jenishkatuwal11/freeCodeCamp into fix-typos-javascript-quiz
